crates/katana/cli/src/args.rs

```diff
@@ -68,6 +69,19 @@ pub struct NodeArgs {
     #[arg(value_parser = katana_core::service::messaging::MessagingConfig::parse)]
     pub messaging: Option<MessagingConfig>,
 
+    /// Disable sampling for the Gas Price Oracle, and fall back on hardcoded gas price values.
+    #[arg(long)]
+    #[arg(conflicts_with = "l1_provider_url")]
+    #[arg(help = "Disable L1 gas sampling and use hardcoded values.")]
+    pub no_sampling: bool,
```

Comment:
```suggestion
```

no need to have dedicated flag to disable sampling. we will just imply it from the cli args. meaning, if user provides the flags for the hardcoded value (ie using `--gpo.l1-eth-gas-price` etc) then we don't have to sample gas prices.

but what i'd like to do here, we still want to maintain the same DX when running katana because if we make sampling as the default, then user would have to either (1) find a eth provider url, or (2) specify the hardcoded gas prices values themselves. both options will only be adding extra frictions for users who are already accustomed to using katana as a devnet before.

so for now, we only need to add a single new arg ie `--l1.provider`. this arg will be important for productionalizing katana later, meaning it will have other usages outside of just gas oracle.